crates/sui-framework/packages/sui-system/sources/staking_pool.move

```diff
@@ -371,8 +375,15 @@ module sui_system::staking_pool {
     /// Called at epoch boundaries to process pending stake withdraws requested during the epoch.
     /// Also called immediately upon withdrawal if the pool is inactive.
     fun process_pending_stake_withdraw(pool: &mut StakingPool) {
-        pool.sui_balance = pool.sui_balance - pool.pending_total_sui_withdraw;
-        pool.pool_token_balance = pool.pool_token_balance - pool.pending_pool_token_withdraw;
+        pool.sui_balance =
```

Comment:
How can you end up in a scenario where the `pending_total_sui_withdraw` is actually more than the balance?

Reply:
check `test_process_pending_stake_withdraw_no_underflow_in_safe_mode` and line 661-666

Reply:
I see, my more precise question is why is this even possible (and why do we accept it as being possible instead of figuring out how to prevent it in the first place?). Accepting it is fine by me, I mainly try to understand why we do.